File: myLeetCode/Hash_Table/349. Intersection of Two Arrays.cpp
// 349. Intersection of Two Arrays
// https://leetcode-cn.com/problems/intersection-of-two-arrays/

#include <iostream>
#include <vector>
#include <unordered_set>

std::vector<int> intersect(std::vector<int>& nums1, std::vector<int>& nums2) {
    std::unordered_set<int> set1(nums1.begin(), nums1.end());
    std::unordered_set<int> set2(nums2.begin(), nums2.end());

    std::vector<int> res;
    for (int i : set1) {
        if (set2.count(i)) {
            res.push_back(i);
            set2.erase(i);
        }
    }

    return res;
}

int main() {
    // std::vector<int> nums1 = {1, 2, 2, 1};
    // std::vector<int> nums2 = {2, 2};
    std::vector<int> nums1 = {4,9,5};
    std::vector<int> nums2 = {9,4,9,8,4};
    std::vector<int> res = intersect(nums1, nums2);
    for (int i : res) {
        std::cout << i << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: C++/代码随想录/Hash_Table/349. Intersection of Two Arrays.py
def intersection(nums1, nums2):
    set1 = set(nums1)
    set2 = set(nums2)
    return list(set1 & set2)

# 测试用例
print(intersection([1, 2, 2, 1], [2, 2]))  # 应该输出 [2]
print(intersection([4, 9, 5], [9, 4, 9, 8, 4]))  # 应该输出 [9, 4]
